analysis/deep2_r23_o32/curvefit_analysis.py

```diff
@@ -270,45 +262,34 @@ def run_experiment_Zcal(fitspath, fitspath_curvefit, fitspath_ini, n_bins=4,
             label='Bian Data Range')
 
     # Next we plot the IDS
-    if include_rlimit:
-        # Detections and Robust limits should always be the first two arrays
-        # in the list of arrays
-        label_IDs = [ID_arrs[0], ID_arrs[1]]
-        for aa in range(len(label_IDs)):
-            for jj in range(len(ID_arrs[aa])):
-                ax.annotate(ID_arrs[aa][jj],
-                            (lR23_arrs[aa][jj], OH_arrs[aa][jj]), fontsize='6')
-    else:
-        for jj in range(len(ID_arrs[0])):
-            ax.annotate(ID_arrs[0][jj], (lR23_arrs[0][jj], OH_arrs[0][jj]),
-                        fontsize='6')
+    for jj in range(len(ID_arrs[0])):
+        ax.annotate(ID_arrs[0][jj], (lR23_arrs[0][jj], OH_arrs[0][jj]),
+                    fontsize='6')
+    for rr in range(len(rlimit_ID)):
+        ax.annotate(rlimit_ID[rr], (rlimit_lR23[rr], rlimit_OH[rr]),
+                    fontsize='6')
+
     ax.set(xlim=(0.0, 1.2))
     ax.set_xlabel(r'$\log(R_{23})$')
     ax.set_ylabel(r'$12+\log({\rm O/H})_{T_e}$')
     # print('PDF file name: ', pdf_file)
     fig.savefig(pdf_file)
     fitting_model = 'Zcal'
     # This section plots makes the plot difference files
+    lR23_arrs = [det_lR23, rlimit_lR23, DEEP2_lR23, MACT_lR23]
+    lO32_arrs = [det_lO32, rlimit_lO32, DEEP2_lO32, MACT_lO32]
+    OH_arrs = [det_OH, rlimit_OH, DEEP2_OH, MACT_OH]
+    ID_arrs = [det_ID, rlimit_ID, DEEP2_id, MACT_ID]
```

Comment:
Wondering if there's a way to easily append `rlimit` since you define lR23_arrs above